### DISCUSSION TITLE
Apps like Fastfetch can't request Bluetooth permission on ghostty's behalf

### DISCUSSION THREAD

**Discussion by evieluvsrainbows** [ghostty-org/ghostty#3995] [Feature Requests, Ideas] [answered]
It appears that Ghostty on macOS lacks support for apps like fastfetch (when ran with "fastfetch -c all") to request the Bluetooth permission on ghostty's behalf, meaning fastfetch abruptly aborts. Screenshots have been attached for ghostty, and the regular macOS Terminal app. I'm not the best at filing issue reports, but I figured I would file this one since it doesn't seem like there are any existing issues filed regarding this.

**Edit**: Manually adding ghostty as an allowed app under Bluetooth in System Settings -> Privacy & Security fixes the issue, but it seems ghostty lacks support for showing the actual permissions prompt currently.

**Edit 2**: A pull request has been opened (#4668) to add the appropriate permission descriptions to Ghostty's application manifest. If this is merged, this will hopefully fix the abort issues, as according to Apple's documentation, these permission description keys are necessary to request if anything being run through these kinds of apps use the technologies or hardware associated with these permission keys.

**Edit 3**: This issue has been confirmed fixed with my PR, the permission dialog for Bluetooth now appears now that my PR has been merged, as seen with the below screenshot.
<img width="372" alt="Screenshot 2025-01-07 at 10 57 56 PM" src="https://github.com/user-attachments/assets/3bd67b1b-d158-46a7-bdf7-2ff6aaf26e4b" />

**ghostty output**:
<img width="650" alt="Screenshot 2024-12-29 at 9 33 14 PM" src="https://github.com/user-attachments/assets/01152b83-8cf0-4ba9-b680-b67842879486" />
**macOS Terminal output**:
<img width="892" alt="image" src="https://github.com/user-attachments/assets/d231571d-5a51-44bf-aacf-4f8649d0edb2" />

### Specs:
* MacBook Pro (November 2023, M3 Max, 48GB RAM)
* macOS 15.2
* ghostty version 1.0.1-main+3059d903

**Comment by CarterLi** [comment]
Fastfetch aborts at [`IOBluetoothDevice.pairedDevices`](https://github.com/fastfetch-cli/fastfetch/blob/dev/src/detection/bluetooth/bluetooth_apple.m#L15). The same issue happens in Warp.

**Reply by evieluvsrainbows** [reply]
Oh. Any possible idea as to why this happens with this and Warp and if its an issue with these specific terminal apps or an issue with the way Fastfetch handles retrieval for paired Bluetooth devices?

**Reply by CarterLi** [reply]
https://github.com/gnachman/iTerm2/blob/master/plists/dev-iTerm2.plist#L137-L138

**Reply by evieluvsrainbows** [reply, marked as answer]
I have opened a Pull Request that adds the necessary permission descriptions, similar to iTerm2, Kitty, and the official Terminal app. If merged, this may fix abort issues.

**Comment by filipwiech** [comment]
Could be related to the #4512 (a similar Mac permissions issue)?

**Comment by evieluvsrainbows** [comment]
Closing for now due to my PR being merged.
